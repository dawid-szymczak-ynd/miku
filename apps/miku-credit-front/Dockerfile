FROM nginx:alpine

WORKDIR /var/www/html
COPY dist/apps/miku-credit-front /var/www/html
COPY apps/miku-credit-front/deployment/nginx.conf /etc/nginx/nginx.conf

EXPOSE 80
