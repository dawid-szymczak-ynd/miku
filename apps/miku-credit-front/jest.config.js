module.exports = {
  name: 'miku-credit-front',
  preset: '../../jest.config.js',
  coverageDirectory: '../../coverage/apps/miku-credit-front',
  snapshotSerializers: [
    'jest-preset-angular/build/AngularNoNgAttributesSnapshotSerializer.js',
    'jest-preset-angular/build/AngularSnapshotSerializer.js',
    'jest-preset-angular/build/HTMLCommentSerializer.js',
  ],
};
