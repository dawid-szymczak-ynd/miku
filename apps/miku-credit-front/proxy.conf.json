{
  "/api": {
    "target": "http://api",
    "secure": false,
    "pathRewrite": {
      "^/api": ""
    }
  }
}
