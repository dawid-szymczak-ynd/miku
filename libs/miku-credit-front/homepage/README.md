# miku-credit-front-homepage

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test miku-credit-front-homepage` to execute the unit tests.
