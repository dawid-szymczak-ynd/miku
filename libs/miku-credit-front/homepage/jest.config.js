module.exports = {
  name: 'miku-credit-front-homepage',
  preset: '../../../jest.config.js',
  coverageDirectory: '../../../coverage/libs/miku-credit-front/homepage',
  snapshotSerializers: [
    'jest-preset-angular/build/AngularNoNgAttributesSnapshotSerializer.js',
    'jest-preset-angular/build/AngularSnapshotSerializer.js',
    'jest-preset-angular/build/HTMLCommentSerializer.js',
  ],
};
