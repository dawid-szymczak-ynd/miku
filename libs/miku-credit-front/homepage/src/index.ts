export * from './lib/miku-credit-front-homepage.module';
