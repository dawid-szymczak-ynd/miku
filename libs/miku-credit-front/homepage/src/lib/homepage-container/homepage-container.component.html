<div class="container">
  <section class="container__right-side">
    <div class="container__right-side__header">
      <button mat-stroked-button [matMenuTriggerFor]="belowMenu" class="container__right-side__header__cta">
        Menu
      </button>
      <mat-menu #belowMenu="matMenu" xPosition="before">
        <button routerLink="/sell-soul/login" mat-menu-item>Log in</button>
      </mat-menu>
    </div>
    <div class="container__right-side__body">
      <div>
        <h1 class="container__right-side__body__title">Let us help!</h1>
        <p class="container__right-side__body__paragraph">
          We will love to help you with taking a loan! Money will appear on your account even in 15 minutes, and it will
          cost only your soul ;)
        </p>
        <button mat-flat-button routerLink="/sell-soul" class="container__right-side__body__cta" data-cy="mainCTA">
          Take your money!
        </button>
      </div>
    </div>
    <div class="container__right-side__footer"></div>
  </section>
  <section class="container__left-side">
    <div>
      <mat-icon class="container__left-side__logo">account_balance</mat-icon>
      <h1 class="container__left-side__title" data-cy="leftTitle">Miku Credit</h1>
      <h2 class="container__left-side__subtitle">Make us fat cat!</h2>
    </div>
  </section>
</div>
