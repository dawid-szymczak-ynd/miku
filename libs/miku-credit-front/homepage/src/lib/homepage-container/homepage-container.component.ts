import { Component } from '@angular/core';

@Component({
  selector: 'miku-credit-homepage-container',
  templateUrl: './homepage-container.component.html',
  styleUrls: ['./homepage-container.component.scss'],
})
export class HomepageContainerComponent {
  constructor() {}
}
