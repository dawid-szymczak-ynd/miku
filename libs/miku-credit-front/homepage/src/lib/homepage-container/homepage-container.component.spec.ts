import { MikuCreditFrontHomepageModule } from '@miku-credit/miku-credit-front/homepage';

import { Shallow } from 'shallow-render';

import { HomepageContainerComponent } from './homepage-container.component';

describe('HomepageContainerComponent', () => {
  let shallow: Shallow<HomepageContainerComponent>;

  beforeEach(() => {
    shallow = new Shallow(HomepageContainerComponent, MikuCreditFrontHomepageModule);
  });

  it('should match snapshot', async () => {
    const { fixture } = await shallow.render();

    expect(fixture).toMatchSnapshot();
  });
});
