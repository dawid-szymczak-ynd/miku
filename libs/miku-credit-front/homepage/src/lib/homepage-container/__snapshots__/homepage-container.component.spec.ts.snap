// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageContainerComponent should match snapshot 1`] = `
<ng-component>
  <miku-credit-homepage-container>
    <div
      class="container"
    >
      <section
        class="container__right-side"
      >
        <div
          class="container__right-side__header"
        >
          <button
            class="container__right-side__header__cta"
            mat-stroked-button=""
          >
             Menu 
          </button>
          <mat-menu
            xposition="before"
          >
            <button
              mat-menu-item=""
              routerlink="/sell-soul/login"
            >
              Log in
            </button>
          </mat-menu>
        </div>
        <div
          class="container__right-side__body"
        >
          <div>
            <h1
              class="container__right-side__body__title"
            >
              Let us help!
            </h1>
            <p
              class="container__right-side__body__paragraph"
            >
               We will love to help you with taking a loan! Money will appear on your account even in 15 minutes, and it will cost only your soul ;) 
            </p>
            <button
              class="container__right-side__body__cta"
              data-cy="mainCTA"
              mat-flat-button=""
              routerlink="/sell-soul"
            >
               Take your money! 
            </button>
          </div>
        </div>
        <div
          class="container__right-side__footer"
        />
      </section>
      <section
        class="container__left-side"
      >
        <div>
          <mat-icon
            class="container__left-side__logo"
          >
            account_balance
          </mat-icon>
          <h1
            class="container__left-side__title"
            data-cy="leftTitle"
          >
            Miku Credit
          </h1>
          <h2
            class="container__left-side__subtitle"
          >
            Make us fat cat!
          </h2>
        </div>
      </section>
    </div>
  </miku-credit-homepage-container>
</ng-component>
`;
